### PR TITLE
VMware: add a new switch allow_raw_data to control raw data feature

### PR DESCRIPTION
**Pull request by xiaofengw-vmware**
…update the documentation. Fix bugs about max_wait.

summary: Introduce a new switch allow_raw_data to control raw data feature, its default value is true. Also fix bugs about max_wait. Update the documentation accordingly.

## Test Steps
Set allow_raw_data to True and False in datasource config, trigger the guest customization on VMware vsphere and verify cloud-init could get the execpted data source.

## Checklist:

 - [ X] My code follows the process laid out in [the documentation](https://cloudinit.readthedocs.io/en/latest/topics/hacking.html)
 - [ X] I have updated or added any unit tests accordingly
 - [ X] I have updated or added any documentation accordingly

### CODE REVIEW

File: cloudinit/sources/DataSourceOVF.py [hunk 1]
```diff
@@ -440,6 +453,8 @@ def get_max_wait_from_cfg(cfg):
 def wait_for_imc_cfg_file(filename, maxwait=180, naplen=5,
                           dirpath="/var/run/vmware-imc"):
     waited = 0
+    if maxwait <= naplen:
+        naplen = 1
```
Comment (TheRealFalcon) [inline cloudinit/sources/DataSourceOVF.py:457]: Nit: would `naplen = maxwait` be better here?

**Review by TheRealFalcon — APPROVED** [review]
One nit inline.

Also, it might be time to start breaking up `_get_data` into smaller methods. I don't think it's necessary for this PR, but going through the layers of nesting is getting hard to read.

**Comment by xiaofengw-vmware** [comment]
> One nit inline.
> 
> Also, it might be time to start breaking up `_get_data` into smaller methods. I don't think it's necessary for this PR, but going through the layers of nesting is getting hard to read.

Thanks, I agree with you. It needs a full test and I prefer to do it in another pull request. I have created an internal PR to address this.

File: cloudinit/sources/DataSourceOVF.py (hunk 1 above)
Comment (xiaofengw-vmware) [reply cloudinit/sources/DataSourceOVF.py:457]: If user set maxwait <= naplen,  it seems he is sensitive to this time delay, so check the cfg file at each seconds might be better. How do you think?

Comment (xiaofengw-vmware) [reply cloudinit/sources/DataSourceOVF.py:457]: @TheRealFalcon

Comment (TheRealFalcon) [reply cloudinit/sources/DataSourceOVF.py:457]: Sure, that makes sense.

**Pull request merged**